lib/generator.js

```diff
@@ -75,7 +75,7 @@ class Generator {
    * @param {String} [options.templateParams]   Optional parameters to pass to the template. Each template define their own params.
    * @param {String} [options.entrypoint]       Name of the file to use as the entry point for the rendering process. Use in case you want to use only a specific template file. Note: this potentially avoids rendering every file in the template.
    * @param {String[]} [options.noOverwriteGlobs] List of globs to skip when regenerating the template.
-   * @param {String[]} [options.disabledHooks] List of hook types to disable.
+   * @param {Object<String, Boolean | String[]>} [options.disabledHooks] Object with hooks to disable. The field is a hook type. If field has "true" value, then generator skips all hooks from given type. If field is array of strings, then generator skips only hooks from array. 
```

Comment:
Best would be if implementation support user to provide just one hook name as string, not array

```suggestion
   * @param {Object<String, Boolean | String | String[]>} [options.disabledHooks] Object with hooks to disable. The key is a hook type. If key has "true" value, then the generator skips all hooks from the given type. If the key is a string with the name of a single hook, then the generator skips only this single hook name. If the key is an array of strings, then the generator skips only hooks from the array. 
```

Reply:
I changed logic, but I have problem with `isHookAvailable` function (https://github.com/asyncapi/generator/blob/master/lib/generator.js#L770). Should it has extended signature `isHookAvailable(hookType, hookName)`? If user passes only hookType then function will behave as before, but when user gives hookName it will check a more extended form?

Reply:
> If user passes only hookType then function will behave as before, but when user gives hookName it will check a more extended form

what do you mean? it is a private function, the user will not use it directly. 
`setFileTemplateName` is a special hook that lets you interact with the process of [file templates](https://github.com/asyncapi/generator/blob/master/docs/authoring.md#file-templates) generation, so you can influence the file name. So it is important to know if `setFileTemplateName` hook is available, not specific name of the hook. Maybe `isHookAvailable` should become `isHookTypeAvailable` or at least jsdoc extended?

Reply:
This function is public :) In JS when you want make method/property private then signature will be `#isHookAvailable`

Reply:
You can check tests here https://github.com/asyncapi/generator/pull/442/files#diff-d2f4f8516ea7f901574dcfd6ad677f6a3bcf85de3606ab5fd2a6a0be241147b3R520

Reply:
Well, it is not public as it is marked as private and not visible in [API docs](https://github.com/asyncapi/generator/blob/master/docs/api.md)

Reply:
Even if this function should behave as private, I think that adapting the function to the new functionality is good idea :)

Reply:
sure, just tell me what is the use case for this new functionality 😄

Reply:
If someone wants to use the generator as a library (not cli) ) then after reading hooks into the registry, person can check if a hook with a given name exists (I know a function is private in JSDoc, but someone can always know about this function). If this use case is not important for you, I can remove the extra code :)

Reply:
```
* @param {Object<String, Boolean | String | String[]>} [options.disabledHooks] Object with hooks to disable. The key is a hook type. If key has "true" value, then the generator skips all hooks from the given type. If the key is a string with the name of a single hook, then the generator skips only this single hook name. If the key is an array of strings, then the generator skips only hooks from the array. 
```

I'm not sure if I'm understanding correctly but seems to me this explanation is wrong. It says: `If the key is a string with the name of a single hook, then the generator skips only this single hook name.  If the key is an array of strings, then the generator skips only hooks from the array. `. I think it should be: `If the value associated with a key is a string with the name of a single hook, then the generator skips only this single hook name.  If the value associated with a key is an array of strings, then the generator skips only hooks from the array. `

Aside from that, I can't really understand what's the use case. @magicmatatjahu you described **how** you would use this feature but I'm more interested in the **why** or **what for**.

Reply:
@fmvilas Thanks! Your description is more verbose. Some people may be confused on reading `If the key is a string with the name of a single hook, then the generator skips only this single hook name.`

I have only one use case in my mind for new signature of `isHookAvailable`. When I will use generator as a library, I would like to check if the specific hook of given hook type is available in current template executing. What does it mean? E.g: I'm using html template, but before render template, I want check if `generatePdf` hook is available (also not disabled). So for this I have `isHookAvailable` function. If hook is disabled/doesn't exist, then I would like perform some additional logic, if yes another logic etc. Of course as I wrote, if this use case is not important I can revert changes to old body of function.

Reply:
> If hook is disabled/doesn't exist, then I would like perform some additional logic, if yes another logic etc.

What logic woudl it be? I think the case here is that in general, better is to add logic that one can back with real world example, otherwise, it is definitely better to not add additional code when not needed

Reply:
I don't see a use case for it being a public function. If you are using the generator as a library instead of the cli you would have access to the provided generator options and therefore the disabled hook right? 😄

Reply:
YOU WIN!!! 😄 I changed to old arguments, but I had to change body to adjust for new `disabledHooks` type -> from array to object.